harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,919 @@
+
+        result = subprocess.check_output(command, shell=False, encoding="utf-8")
+
+        assert result.find(f"64 bytes from {vlan_ip}") > 0, (
+            f"Failed to ping VM external vlan IP {vlan_ip} "
+            f"on vlan interface from external node")
+
+        # SSH to vlan ip address and execute command from external host
+        _stdout, _stderr = self.ssh_client(
+            client, vlan_ip, vm_credential["user"], vm_credential["password"], 'ls', wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            f"Failed to ssh to VM external vlan IP {vlan_ip}"
+            f"on vlan interface from external node")
+
+        # cleanup vm
+        delete_vm(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="reboot_vlan_connection",
+                            depends=["vlan_network_connection"])
+    def test_reboot_vlan_connection(self, api_client, request, unique_name,
+                                    image_info, vlan_network, wait_timeout):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/negative-vlan-after-reboot/
+
+
+        Steps:
+        1. Create an external VLAN network
+        2. Create a new VM already with the vlan network
+        3. Check can ping external VLAN IP
+        4. Reboot VM
+        5. Ping VM during reboot
+        6. Check can't ping VM during reboot
+        7. Check the VM should reboot
+        8. Ping VM during after reboot
+        9. Check can ping VM
+        """
+        unique_name = unique_name + "-reboot-vlan"
+
+        image_url = image_info.url
+
+        # Check image exists
+        code, data = api_client.images.get(image_name)
+
+        if code == 404:
+            create_image_url(api_client, image_name, image_url, wait_timeout)
+
+        spec = api_client.vms.Spec(1, 2, mgmt_network=False)
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+
+        # Create VM
+        spec.add_image(image_name, "default/" + image_name)
+        spec.add_network("default", vlan_network['id'])
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # Get VM interface ipAddresses
+        code, data = api_client.vms.get_status(unique_name)
+        assert 200 == code, (f"Failed to get specific vm content: {code}, {data}")
+
+        interfaces_data = data['status']['interfaces']
+        for interface in interfaces_data:
+            ip_addresses = interface['ipAddresses']
+
+        vlan_ip = ip_addresses[0]
```

Comment:
You didn't check the IP is from the (VLAN) NIC, how can you say it is `vlan_ip`?